### ISSUE TITLE
Migrating old (bitwarden-ruby) database causes attachment-related error

### ISSUE THREAD

**Issue by ZNixian** [jcs/rubywarden#72]
Hello,

Unfortunately I don't have the output from the migration tool (I was ssh'd into my server from a virtual terminal, and didn't think to save the output before fixing it), however it was very similar #63, but on the attachments table - here's what it should have been, based on 63's log:

```
/home/bitwarden/.rvm/gems/ruby-2.4.4/gems/sqlite3-1.3.13/lib/sqlite3/database.rb:91:in `initialize': SQLite3::SQLException: no such table: attachments: SELECT COUNT(*) FROM "attachments" (ActiveRecord::StatementInvalid)
```

I was able to fix this by checking out e539b67 (before the attachments were implemented, according to the commit log) and running the migration tool, which then ran successfully.

I should also note that I was 25 commits behind origin/master when I pulled (head at the time was ce9d0c4). If the cause isn't obvious and you can't easily recreate it, I'll grab my old database and run the migration again, and grab the error.

**Comment by universal** [comment]
The error most likely originates from: https://github.com/jcs/rubywarden/blob/master/tools/migrate_to_ar.rb#L55 

The new code has an attachment model defined, and therefore tries to look for an ```attachments``` table . We should check what classes are defined and what tables are present in the old database and only migrate those that are present in both!

so an intersection of: ```ActiveRecord::Base.connection.tables``` and ```ObjectSpace.each_object(Class).select {|k| k < DBModel}.map {|k| k.table_name }``` . 

so replacing the above mentioned line with something like this probably solves the problem. I added the surrounding lines for context :-)

```ruby
# select only tables for defined models
class YamlDb::SerializationHelper::Dump
  def self.tables
    ActiveRecord::Base.connection.tables & ObjectSpace.each_object(Class).select {|k| k < DBModel}.map {|k| k.table_name }
  end
end
```

If you could check if the adopted code works with your old database, that would be great :-)

**Comment by kniteli** [comment]
@universal 

I followed that advice and it did get past that error, but it looks like there's some other issues with the migration. Specifically, if I'm reading the migrations right (I don't really know ruby so I may be wrong), it looks like you normalized the attachments into their own table, but when you dump the old database and import the new you're just flatly importing it with no mapping. Obviously there's no longer an attachments column so the import fails.

Luckily I had no attachments and not too many items in my database, so I was able to manually delete the attachments column from the yaml dump file, after which the import proceeded normally and my rubywarden install works again.
